tests/test_ops.py

```diff
@@ -193,6 +193,42 @@ def test_stratify(self):
                     f"({original_name}) to stratified compartment ({key})"
             )
 
+    def test_stratify_w_name_map(self):
+        """Test stratifying a template model by labels."""
+        city_name_map = {
+            "geonames:5128581": "New York City", "geonames:4930956": "Boston"
+        }
+        actual = stratify(
+            sir_parameterized,
+            key="city",
+            strata=cities,
+            strata_name_map=city_name_map,
+            cartesian_control=False,
+            directed=False
+        )
+        for template in actual.templates:
+            for concept in template.get_concepts():
+                self.assertIn("city", concept.context)
+        self.assert_unique_controllers(actual)
+        self.assertEqual(
+            {template.get_key() for template in sir_2_city.templates},
+            {template.get_key() for template in actual.templates},
+        )
+
+        original_name = "susceptible_population"
+        self.assertIn(original_name, sir_parameterized.initials)
+        for city in cities:
+            city_name = city_name_map.get(city, city)
```

Comment:
can you write out what these strings are explicitly in a dictionary above rather than generating them procedurally?

Reply:
Resolved in 567f90ce4c9bd50705d9dbcc327d5d41077fff7f